com.unity.shadergraph/Editor/Data/SubGraph/SubGraphAsset.cs

```diff
@@ -47,6 +48,7 @@ class SubGraphAsset : ScriptableObject, ISerializationCallbackReceiver
         public string hlslName;
 
         public string assetGuid;
+        internal string subGraphObjectIdOverride; // allow id to be overwritten when necessary
```

Comment:
Would it make more sense to pass the "SubGraphData object id override" as a parameter on WriteData()?

Reply:
It could, I was just following the paradigm above, where the assetGuid is exposed directly and provided during the import process above.